eli5/sklearn/treeinspect.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+"""
+Inspect scikit-learn decision trees.
+
+This is an alternative to sklearn.tree.export which doesn't require graphviz
+and provides a way to output result in text-based format.
+"""
+from __future__ import absolute_import, division
+from sklearn.tree import _tree
+
+
+def tree2dict(decision_tree, feature_names=None):
+    """
+    Convert DecisionTreeClassifier or DecisionTreeRegressor
+    to an inspectable dictionary.
+    """
+    res = {
+        "criterion": decision_tree.criterion,
+        "tree": _node2dict(decision_tree.tree_, 0),
+    }
+    _add_feature_names(res, feature_names)
+    return res
+
+
+def _add_feature_names(treedict, feature_names=None):
+    for node in _treeiter(treedict['tree']):
+        if not node['is_leaf']:
+            feat_id = node['feature_id']
+            if feature_names is None:
+                node['feature_name'] = "x%s" % feat_id
+            else:
+                node['feature_name'] = feature_names[feat_id]
+
+
+def _node2dict(tree, node_id):
+    is_leaf = tree.children_left[node_id] == _tree.TREE_LEAF
```

Comment:
the code above is a way to check if a current node is a leaf, not if its left child is a leaf; I copy-pasted it from export_graphviz